hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -754,12 +758,12 @@ protected void bootstrapCommit(List<DirectoryInfo> partitionInfoList, String cre
     // Is this a hoodie partition
     private boolean isHoodiePartition = false;
 
-    public DirectoryInfo(String relativePath, FileStatus[] fileStatus) {
+    public DirectoryInfo(final HoodieTableMetaClient datasetMetaClient, String relativePath, FileStatus[] fileStatus) {
       this.relativePath = relativePath;
 
       // Pre-allocate with the maximum length possible
       filenameToSizeMap = new HashMap<>(fileStatus.length);
-
+      final HoodieTimeline completedInstantsTimeline = datasetMetaClient.getActiveTimeline().filterCompletedInstants();
```

Comment:
we can do the filtering outside and pass it in here.

Reply:
I am taking on this new functionality change in a new PR